client/my-sites/upgrades/domain-management/dns/txt-record.jsx

```diff
@@ -57,7 +64,12 @@ const TxtRecord = React.createClass( {
 						onChange={ this.props.onChange }
 						value={ this.props.fieldValues.data }
 						placeholder={ this.translate( 'e.g. %(example)s', { args: { example: 'v=spf1 include:example.com ~all' } } ) } />
-					{ ! isDataValid ? <FormInputValidation text={ this.translate( 'Invalid TXT Record' ) } isError={ true } /> : null }
+					{ hasNonAsciiData && (
+						<div className="form-input-validation dns__warning">
```

Comment:
How about extending `FormInputValidation` to support three severities? `success`, `warning` and `error` (maybe even `info`, too, for good measure?). That'd make this change much more straightforward and "clean". You could keep the `isError` property for backwards compatibility (changing all uses of `FormInputValidation` is way out of scope of this PR).